assets/src/stories-editor/components/block-mover/block-draggable.js

```diff
@@ -10,28 +10,87 @@ import PropTypes from 'prop-types';
 /**
  * WordPress dependencies
  */
-import { withSelect } from '@wordpress/data';
+import {
+	withSelect,
+	dispatch,
+} from '@wordpress/data';
+import { cloneBlock } from '@wordpress/blocks';
+import { useRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Draggable from './draggable';
+import { getPercentageFromPixels } from '../../helpers';
 
-const BlockDraggable = ( { children, clientId, blockName, rootClientId, blockElementId, index, onDragStart, onDragEnd } ) => {
+const BlockDraggable = ( { children, clientId, blockName, rootClientId, blockElementId, block, index, getBlockOrder, onDragStart, onDragEnd } ) => {
 	const transferData = {
 		type: 'block',
 		srcIndex: index,
 		srcRootClientId: rootClientId,
 		srcClientId: clientId,
 	};
 
+	const { setCurrentPage } = dispatch( 'amp/story' );
+	const { selectBlock, removeBlock, insertBlock, updateBlockAttributes } = dispatch( 'core/block-editor' );
+
+	const getNeighborPageId = ( offset ) => {
+		const pages = getBlockOrder();
+		const currentPageIndex = pages.findIndex( ( i ) => i === rootClientId );
+		const newPageIndex = currentPageIndex + offset;
+
+		// Do we even have a neighbor in that direction? (offset=0 is not a neighbor)
+		if ( offset === 0 || newPageIndex < 0 || newPageIndex >= pages.length ) {
+			return null;
+		}
+
+		return pages[ newPageIndex ];
+	};
+
+	const onNeighborDrop = ( offset, finalPosition ) => {
```

Comment:
At first glance, I'd probably use `withDispatch` and put `onNeighborDrop` and `onNeighborHover`  there.

Reply:
Hm, that's not really doable due to the `currentHoverElement` as it's created with a hook not available in a `withSelect` handler (I think?). But maybe it's not even necessary to create this element with a hook, as the BlockDraggable is never re-rendered while the drag is ongoing (at least I don't think it is). Will try to experiment with this.

Reply:
The hook isn't necessary in the component, but it definitely can't go in a `withDispatch` as it's re-invoked for every drag of the element (as the register updates I guess).

I'll put another component in-between and see if I like the result.